src/app/base/sagas/websockets.ts

```diff
@@ -352,6 +357,25 @@ export function* handleUnsubscribe(
   }
 }
 
+// List of websocket response errors triggering connection failure
+const WEBSOCKET_RESPONSE_CONNECTION_ERRORS = [
+  "Vault connection failed",
+  "Vault request failed",
+];
+
+export function* handleResponseError(
+  response: WebSocketResponseError
+): SagaGenerator<void> {
+  if (WEBSOCKET_RESPONSE_CONNECTION_ERRORS.includes(response.error)) {
+    yield* put({
+      error: true,
+      payload: response.error,
+      type: "status/websocketError",
+    });
+  } else {
+    yield* put(messageActions.add(response.error, "negative"));
+  }
```

Comment:
@ndv99  Special case for Vault errors (it's a bit of a hack that we should probably avoid).